packages/@eventual/aws-cdk/src/bucket-service.ts

```diff
@@ -0,0 +1,287 @@
+    // find any bucket names that were provided by the service and not computed
+    const bucketNameOverrides = this.props.bucketOverrides
+      ? Object.values(
+          this.props.bucketOverrides as Record<string, BucketRuntimeOverrides>
+        )
+          .map((s) => s.bucketName)
+          .filter((s): s is string => !!s)
+      : [];
+
+    // grants the permission to start any task
+    grantee.grantPrincipal.addToPrincipalPolicy(
+      new PolicyStatement({
+        actions: ["s3:List*"],
+        resources: [
+          serviceBucketArn(
+            this.props.serviceName,
+            bucketServiceBucketSuffix("*"),
+            false
+          ),
+          ...bucketNameOverrides.map(formatBucketArn),
+        ],
+      })
+    );
+
+    grantee.grantPrincipal.addToPrincipalPolicy(
+      new PolicyStatement({
+        actions: [
+          "s3:List*",
+          "s3:GetObject*",
+          "s3:PutObject*",
+          "s3:DeleteObject*",
+        ],
+        resources: [
+          `${serviceBucketArn(
+            this.props.serviceName,
+            bucketServiceBucketSuffix("*"),
+            false
+          )}/*`,
+          ...bucketNameOverrides.map((s) => formatBucketArn(`${s}/*`)),
+        ],
+      })
+    );
+  }
+
+  private readonly ENV_MAPPINGS = {
+    [ENV_NAMES.SERVICE_NAME]: () => this.props.serviceName,
+    [ENV_NAMES.BUCKET_OVERRIDES]: () =>
+      Stack.of(this.props.serviceScope).toJsonString(
+        this.props.bucketOverrides
+      ),
+  } as const;
+
+  private addEnvs(func: Function, ...envs: (keyof typeof this.ENV_MAPPINGS)[]) {
+    envs.forEach((env) => func.addEnvironment(env, this.ENV_MAPPINGS[env]()));
+  }
+}
+
+interface BucketProps {
+  serviceProps: BucketServiceProps<any>;
+  bucketService: BucketService<any>;
+  bucket: BucketRuntime;
+}
+
+export class Bucket extends Construct {
+  public bucket: s3.Bucket;
+  public handlers: Record<string, BucketNotificationHandler>;
+
+  constructor(scope: Construct, props: BucketProps) {
+    super(scope, props.bucket.name);
+
+    const bucketOverrides =
+      props.serviceProps.bucketOverrides?.[props.bucket.name];
+
+    this.bucket = new s3.Bucket(this, "Bucket", {
+      cors:
+        props.serviceProps.cors &&
+        props.serviceProps.cors.allowMethods &&
+        props.serviceProps.cors.allowOrigins
+          ? [
+              {
+                allowedMethods: props.serviceProps.cors.allowMethods.flatMap(
+                  (s) =>
+                    Object.values(s3.HttpMethods).find(
+                      (v) => s.toString() === v.toString()
+                    ) ?? []
+                ),
+                allowedOrigins: props.serviceProps.cors.allowOrigins,
+                allowedHeaders: props.serviceProps.cors.allowHeaders,
+                exposedHeaders: props.serviceProps.cors.exposeHeaders,
+                maxAge: props.serviceProps.cors.maxAge?.toSeconds(),
+              },
+            ]
+          : undefined,
```

Comment:
Curious, why was this needed?

Reply:
Pre Signed url needs cors.

Reply:
So it was logical to give the same cors that the api has access to. If there are more use cases, we can add overrides later.